File: Part1_강의자료/Ch05_정렬/문제별코드/1931_회의실배정/1931.java
import java.util.*;

class Meeting {
    public Meeting(int start, int end) {
        this.start = start;
        this.end = end;
    }
    int start;
    int end;
}

class Main
{
    public static void main (String[] args) {
        Scanner sc = new Scanner(System.in);

        int N = sc.nextInt();
        Meeting[] meetings = new Meeting[N];
        for (int i = 0; i < N; i++)
            meetings[i] = new Meeting(sc.nextInt(), sc.nextInt());

        Arrays.sort(meetings, (o1, o2) -> {
            if (o1.end == o2.end)
                return o1.start - o2.start;
            return o1.end - o2.end;
        });

        int cnt = 0, ended = 0;
        for (int i = 0; i < N; i++)
            if (ended <= meetings[i].start) {
                cnt++;
                ended = meetings[i].end;
            }
        System.out.println(cnt);
    }
}

File: Part1_강의자료/Ch05_정렬/문제별코드/1931_회의실배정/1931.cpp
#include <algorithm>
#include <stdio.h>
#include <vector>

using namespace std;

struct Meeting {
    int start;
    int end;
};

int main()
{
    int N;
    
    scanf("%d", &N);
    vector<Meeting> meetings(N);

    for (int i = 0; i < N; i++)
        scanf("%d %d", &meetings[i].start, &meetings[i].end);
        
    sort(meetings.begin(), meetings.end(), [](const Meeting& a, const Meeting& b) {
        return a.end == b.end ? a.start < b.start : a.end < b.end;
    });

    int end_time = 0, res = 0;
    for (int i = 0; i < N; i++)
    {
        if (end_time <= meetings[i].start) 
        {
            res++;
            end_time = meetings[i].end;
        }
    }

    printf("%d\n", res);
    return 0;
}
